### ISSUE TITLE
ArgoverseForecastingLoader issues - KeyError and hanging

### ISSUE THREAD

**Issue by aaronh65** [argoai/argoverse-api#84]
I've encountered two issues with the ArgoverseForecastingLoader.

The first issue occurs when I instantiate `afl=ArgoverseForecastingLoader(train_dir)` and try to call `afl[0].city`. The JupyterLab cell will hang on this command, whereas calling `afl[0].agent_traj` works. However, I can get `afl[0].city` to work if I stop the cell during execution and run the cell again.

The second issue occurs as I try to iterate through `afl` to process data. As I iterate through the data, I call `afl[idx].city` and `afl[idx].agent_traj`. However, I encounter a KeyError with `afl[idx].city` at seemingly random indices (see below).
<img width="676" alt="Screen Shot 2019-11-10 at 5 57 36 PM" src="https://user-images.githubusercontent.com/8619969/68552444-1dc40080-03e5-11ea-9683-ef51b51bc67d.png">

Every time this occurs, I can manually confirm that the file described in the KeyError exists at the location specified.

Please advise.

**Comment by jagjeet-singh** [comment]
Hi, Thanks for raising the issue. I have a couple of followup questions:
1. Are you using Argoverse1.1?
2. Are you using forecasting sample dataset?

**Comment by jagjeet-singh** [comment]
I guess both the above questions are answered by your output.
tl;dr: Forecasting data loader is expected to be used when you are using forcasting sample dataset or a smaller subset of train/val/test sets.  https://github.com/argoai/argoverse-api/issues/46

You are using train dataset which has ~250k sequences. Reading csv files for all these sequences will take a long time. `city` and `track_id_list` property

**Comment by jagjeet-singh** [comment]
I've optimized the dataloader to some via this PR
https://github.com/argoai/argoverse-api/pull/86/files

Hope this helps.
